src/BipartiteGraph.java

```diff
@@ -0,0 +1,192 @@
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.util.*;
+
+public class BipartiteGraph
+{
+
+    private int m_numberOfRows;
+    private int m_numberOfColumns;
+    private String m_optimalGenderPrefix;
+    private String m_otherGenderPrefix;
+
+    HashMap<String, String> m_otherGenderEngagementPartner;
+
+    LinkedList<String> m_optimalGenderQueue;
+
+    HashMap<String, LinkedList<String>> m_otherGenderPreferenceHashMap;
+
+    HashMap<String, LinkedList<String>> m_optimalGenderProposalList;
+
+    public BipartiteGraph(
+            int[][] optimalGenderDataSet2DArray,
+            int[][] otherGenderDataSet2DArray,
+            int numberOfRowsAndColumns,
+            String preferenceString)
+    {
+        // Populate data structures
+        // Assumption: square matrix is supplied
+        m_numberOfRows = numberOfRowsAndColumns;
+        m_numberOfColumns = numberOfRowsAndColumns;
+
+        if(0 == preferenceString.compareToIgnoreCase("m"))
+        {
+            // man optimal matching
+            m_optimalGenderPrefix = new String("m");
+            m_otherGenderPrefix = new String("w");
+        }
+        else if (0 == preferenceString.compareToIgnoreCase("w"))
+        {
+            // woman optimal matching
+            m_optimalGenderPrefix = new String("w");
+            m_otherGenderPrefix = new String("m");
+        }
+        else
+        {
+            System.out.println("Invalid Argument: " + preferenceString);
+            System.exit(-100);
+        }
+
+        // I want a stack of preferences for each optimal
+
+        m_optimalGenderProposalList = new HashMap<String, LinkedList<String>>();
+
+        m_otherGenderPreferenceHashMap = new HashMap<String, LinkedList<String>>();
+
+        m_otherGenderEngagementPartner = new HashMap<String,String>();
+
+        // Initialize Queue to include all bidders
+        m_optimalGenderQueue = new LinkedList<String>();
+
+
+        for(int optimalGenderIndex = 0; optimalGenderIndex < m_numberOfRows; ++optimalGenderIndex)
+        {
+            String optimalGenderName = new String(m_optimalGenderPrefix + Integer.toString(optimalGenderIndex + 1));
```

Comment:
Same here. I think the `+` operator actually calls `Integer.toString()` whenever is trying to append a number to a String.
Again, no need to change it, just FYI. Looks cleaner ;)
```java
String optimalGenderName = m_optimalGenderPrefix + (optimalGenderIndex + 1);
```